localstack/services/sns/resource_providers/aws_sns_topic.py

```diff
@@ -0,0 +1,192 @@
+            if attributes.get("FifoTopic") is not None:
+                attributes["FifoTopic"] = str(attributes.get("FifoTopic"))
+
+            if attributes.get("ContentBasedDeduplication") is not None:
+                attributes["ContentBasedDeduplication"] = str(
+                    attributes.get("ContentBasedDeduplication"))
+
+            subscriptions = []
+            if attributes.get("Subscription") is not None:
+                subscriptions = attributes["Subscription"]
+                del attributes["Subscription"]
+
+            # in case cloudformation didn't provide topic name
+            if model.get("TopicName") is None:
+                model["TopicName"] = f"topic-{short_uid()}"
+
+            create_sns_response = sns.create_topic(Name=model["TopicName"],
+                                                   Attributes=attributes
+                                                   )
+            request.custom_context[REPEATED_INVOCATION] = True
+            model["TopicArn"] = create_sns_response["TopicArn"]
+
+            # now we add subscriptions if they exists
+            # TODO do we need to keep arns for unsubscribing before deleting topic
+            for subscription in subscriptions:
+                sns.subscribe(
+                    TopicArn=model["TopicArn"],
+                    Protocol=subscription["Protocol"],
+                    Endpoint=subscription["Endpoint"]
+                )
+
+            return ProgressEvent(
+                status=OperationStatus.SUCCESS,
+                resource_model=model,
+                custom_context=request.custom_context,
+            )
+
+        # - if finished, update the model with all fields and return success event:
+        #   return ProgressEvent(status=OperationStatus.SUCCESS, resource_model=model)
+        # - else
+        #   return ProgressEvent(status=OperationStatus.IN_PROGRESS, resource_model=model)
+
+        return ProgressEvent(status=OperationStatus.SUCCESS, resource_model=model)
+
+    def read(
+            self,
+            request: ResourceRequest[SNSTopicProperties],
+    ) -> ProgressEvent[SNSTopicProperties]:
+        """
+        Fetch resource information
+
+        IAM permissions required:
+          - sns:GetTopicAttributes
+          - sns:ListTagsForResource
+          - sns:ListSubscriptionsByTopic
+          - sns:GetDataProtectionPolicy
+        """
+        raise NotImplementedError
+
+    def delete(
+            self,
+            request: ResourceRequest[SNSTopicProperties],
+    ) -> ProgressEvent[SNSTopicProperties]:
+        model = request.desired_state
+        sns = request.aws_client_factory.sns
+        sns.delete_topic(TopicArn=model["TopicArn"])
+        """
+        Delete a resource
+
+        IAM permissions required:
+          - sns:DeleteTopic
+        """
+        return ProgressEvent(status=OperationStatus.SUCCESS, resource_model={})
```

Comment:
Would be great if you could implement the "deletion" (+ potential unsubscribe) here.

If you're feeling particularly motivated, you can of course also cover this via a test, because I think this is probably not properly covered right now :grimacing:

Reply:
Updated the code.

Test 
`tests.aws.services.cloudformation.resources.test_sns.test_deploy_stack_with_sns_topic`
checks if topic is properly deleted after stack is destroyed.